### PR TITLE
[CFDP-1067] Profile image compression

### PR DESCRIPTION
**Pull request by asleepingpanda**
### 1. What does this PR do, or why is it needed?
This PR establishes the first follow up to image caching since we first introduced Cloudinary. This image enhancement focuses on user's Profile Images.

The following compressions were included (per the available options of [Image Resizer](https://imageresizing.net/docs/v4/docs/basics)):
1. Forced size to 150x150
2. Image quality set to 70 (see documentation)
3. Convert image to jpeg
4. Crop the image when resizing

### 2. What design trade-offs/decisions were made?
With profile images, we had 2 opportunities for image compression: at user upload, at request of the image. With this update, we decided to make the update at the API level to be a request compression.

### 3. How do I test this PR?
Request a profile image from the visual gql interface and look for a url with the following query params:
`https:cloudfront.christfellowship.church/GetImage.ashx?guid=_GUID_&mode=crop&h=150&w=150&format=jpg&quality=75`

I don't fully know the validity of this report, but I ran a speed test on this [image testing website](https://imagespeedtest.piio.co/) and got the following results

### 0.3 second load time
https://cloudfront.christfellowship.church/GetImage.ashx?guid=3376aa0d-5610-4a8a-ae24-046250ebf297&mode=crop&h=150&w=150&format=jpg&quality=70

### 0.7 second load time
https://cloudfront.christfellowship.church/GetImage.ashx?guid=3376aa0d-5610-4a8a-ae24-046250ebf297

Weird that the larger image loaded quicker, but I think the load time is the most important metric.

### 4. What automated tests did you write?
Wrote unit tests for the helper method `rockImageUrl` which is a helper method to be used in the future for generating Rock Url's with Image Resizer options.

## SCREENSHOTS/GIFS

Compressed Image Demo
Screen Shot 2020-12-15 at 9 20 59 AM

## TODO:

- [x] PR has a relevant title that will be understandable in a public changelog (ie...non developers)
- [ ] Closes [tag-issues-here]
- [ ] No new warnings in tests, storybook, or apps
- [x] Assign relevant reviewers
- [x] Upload GIF(s) of iOS, Android, Web
- [x] Get one peer approval before merging

## REVIEW:

**Manual QA**

- [ ] QA branch on iOS and ensure it looks/behaves as expected
- [ ] QA branch on Android and ensure it looks/behaves as expected
- [ ] QA branch on Web and ensure it looks/behaves as expected

**Code Review**

- [ ] Read through the "Files changed" tab _very carefully_
- [ ] Edge cases: what assumptions are made about input?
- [ ] What kind of tests could be written?
- [ ] How might we make this easier for someone else to understand?
- [ ] Could the code be simpler?
- [ ] Will the code be easy to modify in the future?
- [ ] What's one part of these changes that makes you excited to merge it?

> The purpose of PR Review is to _improve the quality of the software._

### CODE REVIEW

**Review by rkdavidson — APPROVED** [review]
This seems like a big performance improvement!
Tested locally and objectively and subjectively feels much faster. Almost feels like there's room to bump the quality back up some, or possibly leave it at 70% but more retina-friendly resolution like ~256x256 or so?

Either way, this feels like a nice surgical benefit across the board 👍.

Left an optional code comment.

**Review by jordanwade — APPROVED** [review]

File: src/data/people/resolver.js [hunk 1]
```diff
@@ -1,138 +1,145 @@
...
-                key: get(ApollosConfig, 'ROCK_MAPPINGS.PERSON_ATTRIBUTES.SALVATION_DATE')
-            })),
+      return phoneNumber ? get(phoneNumber, 'number', '') : '';
+    }),
+    photo: (root, args, { dataSources }) => {
+      const guid = get(root, 'photo.guid');
 
+      return {
+        uri: rockImageUrl(
+          guid && guid !== '' ? guid : '0ad7f78a-1e6b-46ad-a8be-baa0dbaaba8e',
```
[the remark this answers is not in the document]
Comment (asleepingpanda) [reply src/data/people/resolver.js:72]: I'm gonna leave it as a lower quality for the time being and wait for someone to tell me that they want the quality increased.

Nice little lodash suggestion! I'll update that real quick
